migrate_lifecycle.go

```diff
@@ -90,3 +90,12 @@ func (o MigrationOptions) EntriesPath() func(uint64, uint8) string {
 func (o MigrationOptions) Followers() []func(context.Context, LogReader) {
 	return o.followers
 }
+
+func (o *MigrationOptions) WithAntispam(as Antispam) *MigrationOptions {
```

Comment:
This one would very much benefit from a brief comment to highlight that this doesn't prevent duplicate entries being added. This feature is intended to be used to pre-populate a DB for anti-spam, which likely only makes sense if you will transition this log to be an append-only log in the future. If you're strictly mirroring for ever and ever, then no need to use this.